pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -450,7 +451,7 @@ func buildMgmtPortListeners(managementPorts model.PortList, managementIP string)
 // TODO: move to plugins
 // buildInboundAuth adds TLS to the listener if the policy requires one.
 func buildSidecarListenerTLSContext(authenticationPolicy *authn.Policy) *auth.DownstreamTlsContext {
-	if requireTLS, mTLSParams := model.RequireTLS(authenticationPolicy); requireTLS {
+	if requireTLS, mTLSParams := authn_plugin.RequireTLS(authenticationPolicy); requireTLS {
```

Comment:
I think this entire code needs to move into the authN plugin under the "onInboundListener" callback. Hence the todo above the function :). We are already invoking the callbacks. So simply moving this code into the callback should be sufficient.

Reply:
I've tried to implement plugin. One thing I'm not sure is we will need to modify the TlsContext in the filter chains, which seems violate the design for plugin (only append), not to mention need some assumption on what filter chain (it could be more than one?) to modify. Do you think we need to change the interface somehow?

Reply:
Plugin is not append only.. See the example I put in for the onOutboundCluster, where I modify the generated cluster. I don't understand the filterchain part..

Reply:
Actually, I don't understand how exactly the plugin works.

From [this](https://github.com/istio/istio/blob/4e75c5e8a21f62cc2246766eef7fd131c6253a1b/pilot/pkg/networking/core/v1alpha3/listener.go#L261), looks like mutable object have pointer to TCP and HTTP filter only, which it can/should only append (from comment on the [CallbackListenerMutableObjects](https://github.com/istio/istio/blob/bf18b1fa0a7393b7f524f27c2bc06c9356f7caac/pilot/pkg/networking/plugin/plugin.go#L55))  (when do we have Listerner object passed in?). 

For SSL, we will need to modify the [TlsContext](https://github.com/istio/istio/blob/4e75c5e8a21f62cc2246766eef7fd131c6253a1b/pilot/pkg/networking/core/v1alpha3/listener.go#L685), which seems not doable in a clean way.

Any suggestion?

+ @ostromart

Reply:
FYI, I remove the plugin code change from this PR. Will do it in different PR.